sklearn/cluster/k_means_.py

```diff
@@ -1310,8 +1309,8 @@ def fit(self, X, y=None):
 
         Parameters
         ----------
-        X : array-like, shape = [n_samples, n_features]
-            Coordinates of the data points to cluster
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Training instances to cluster
```

Comment:
For now we can have it as

```
Training data to cluster.
```

A more general docstring which is standardized across the codebase will be addressed by fixing #3791 later...